src/d/actor/d_a_obj_mknjd.cpp

```diff
@@ -3,216 +3,1144 @@
+    mMsgInstId = 0xFFFFFFFF;
+    m0504 = false;
+
+    return 1;
 }
 
 /* 000008E8-00000A84       .text Mthd_Create__Q210daObjMknjD5Act_cFv */
-void daObjMknjD::Act_c::Mthd_Create() {
+s32 daObjMknjD::Act_c::Mthd_Create() {
     /* Nonmatching */
-}
+    s32 phase_state;
 
-/* 00000A84-00000B00       .text __dt__18dPa_smokeEcallBackFv */
-dPa_smokeEcallBack::~dPa_smokeEcallBack() {
-    /* Nonmatching */
-}
+    fopAcM_SetupActor(this, daObjMknjD::Act_c);
 
-/* 00000B00-00000B24       .text __defctor__18dPa_smokeEcallBackFv */
-void dPa_smokeEcallBack::__defctor() {
-    /* Nonmatching */
-}
+    m043E = prm_get_Type();
 
-/* 00000B24-00000B28       .text __ct__4cXyzFv */
-cXyz::cXyz() {
-    /* Nonmatching */
-}
+    int switchIdx = prm_get_swSave();
+    if (fopAcM_isSwitch(this, switchIdx)) {
+        mEmitters[2] = NULL;
+        mEmitters[3] = NULL;
 
-/* 00000B28-00000B64       .text __dt__4cXyzFv */
-cXyz::~cXyz() {
-    /* Nonmatching */
+        return 3;
+    }
+    else {
+        phase_state = dComIfG_resLoad(&mPhs, M_arcname);
+        if (phase_state == cPhs_COMPLEATE_e) {
+            phase_state = MoveBGCreate(M_arcname, ID_COL_DZB, NULL, 0x65A0);
+
+            JUT_ASSERT(0x1CA, (phase_state == cPhs_COMPLEATE_e) || (phase_state == cPhs_ERROR_e))
+        }
+    }
+
+    return phase_state;
 }
 
 /* 00000B64-00000BDC       .text Delete__Q210daObjMknjD5Act_cFv */
-void daObjMknjD::Act_c::Delete() {
-    /* Nonmatching */
+int daObjMknjD::Act_c::Delete() {
+    dComIfGp_getAttention().mFlags &= 0x7fffffff;
+
+    for (int i = 0; i < 4; i++) {
+        mSmokeCBs[i].end();
+    }
+
+    return 1;
 }
 
 /* 00000BDC-00000C34       .text Mthd_Delete__Q210daObjMknjD5Act_cFv */
-void daObjMknjD::Act_c::Mthd_Delete() {
-    /* Nonmatching */
+s32 daObjMknjD::Act_c::Mthd_Delete() {
+    int bgDeleteResult = MoveBGDelete();
+    
+    if (mBase.mUnk2 != 3) {
+        dComIfG_resDelete(&mPhs, M_arcname);
+    }
+
+    return bgDeleteResult;
 }
 
 /* 00000C34-00000CC8       .text set_mtx__Q210daObjMknjD5Act_cFv */
 void daObjMknjD::Act_c::set_mtx() {
     /* Nonmatching */
+    mDoMtx_stack_c::transS(getPosition());
+    mDoMtx_stack_c::ZXYrotM(shape_angle);
+
+    mMainMdl->setBaseTRMtx(mDoMtx_stack_c::get());
+    mBreakMdl->setBaseTRMtx(mDoMtx_stack_c::get());
+
+    mDoMtx_copy(mDoMtx_stack_c::get(), M_tmp_mtx);
 }
 
 /* 00000CC8-00000D28       .text init_mtx__Q210daObjMknjD5Act_cFv */
 void daObjMknjD::Act_c::init_mtx() {
     /* Nonmatching */
+    mMainMdl->setBaseScale(cXyz(1.0f, 1.0f, 1.0f));
+    mBreakMdl->setBaseScale(cXyz(1.0f, 1.0f, 1.0f));
+
+    set_mtx();
 }
 
 /* 00000D28-00000E04       .text setGoal__Q210daObjMknjD5Act_cFi */
-void daObjMknjD::Act_c::setGoal(int) {
+void daObjMknjD::Act_c::setGoal(int i_staffIdx) {
     /* Nonmatching */
+    cXyz pos = *dComIfGp_evmng_getMyXyzP(i_staffIdx, "Posion");
+
+    mDoMtx_stack_c::transS(getPosition());
+    mDoMtx_YrotM(mDoMtx_stack_c::get(), current.angle.GetY());
```

Comment:
this can just be `current.angle.y`, the getters didnt exist in original and should prob be removed